docs/DeveloperGuide.md

```diff
@@ -8,8 +8,189 @@
+
+For existing Timetables, they can be used to add and delete Lessons. Each Timetable is made up of a HashMap of ArrayLists(type: Lesson) indexed by weekday e.g. Monday for easy compartmentalisation.
+Within Timetables, Lessons will compromise details like the day, start time and end time. The ArrayLists(type: Lesson) are sorted in non-descending order by the start time of the Lessons.
+
+Relevant exceptions are thrown when unanticipated scenarios occur. For instance, if the user attempts to delete a non-existent Timetable, a TimetableNotFoundException is thrown.
+Another example is the user attempting to add a lesson that is conflicting with an existing lesson in his/her timetable. In this case, a TimetableClashException is thrown.
+
+The following diagram illustrates the relationships between the three main timetable classes - TimetableManager, Timetable and Lesson.
+
+![Timetable Class Diagram](./images/Timetable_Class.png)
+
+In chronological order, the following diagrams illustrate the flow of the program for adding lessons, deleting lessons and displaying timetables to the user.
+
+![Timetable Add Lesson Sequence Diagram](./images/Timetable_addLesson_Sequence.png)
+![Timetable Delete Lesson Sequence Diagram](./images/Timetable_deleteLesson_Sequence.png)
+![Timetable Print Timetable Sequence Diagram](./images/Timetable_printTimetable_Sequence.png)
+
+### Delete History
+
+To help users recall the modules that they had recently deleted, the Delete History feature allows them to view up to 5 most
+recently deleted module mappings. This can help them to add the module mappings back to the lists without having to search for the specific
+module code again. 
+
+The UserDeletedModules class has an ArrayDeque, which stores the recently deleted module mappings.
+When the user deletes a module mapping, it will be added to the ArrayDeque.
+If the ArrayDeque already contains 5 module mappings, the last one (least recent) will be deleted, before the addition of a new module mapping.
+Do note that delete history information is not stored upon exiting the app (ie. it is not stored in User Storage).
+
+The following diagram illustrates the relationship between UserUniversityListManager and UserDeletedModules classes.
+
+![User Deleted Modules Class Diagram](./images/UserDeletedModules_Class.png)
+
+The following diagram illustrates the flow of the program, when a user deletes a module. 
+
+![User Deleted Modules Sequence Diagram](./images/UserDeletedModules_Sequence.png)
+
+### Ui
+
+The Ui class is the cornerstone of the Duke program to facilitate interaction with the user. It is used to scan and collect user input, print error messages to the user upon invalid input commands, 
+and display the appropriate acknowledgements or required information based on the user's command.
+
+The following diagram illustrates the methods within the Ui class that can be invoked by the other classes in Duke for the purpose of user interaction.
```

Comment:
Nice looking class diagram with clearly written functions. I think you might want to consider adding like how the class is connected to other classes in the application, rather than showing it as a standalone UI class on its own